ext/mysqli/mysqli_api.c

```diff
@@ -1513,7 +1513,10 @@ PHP_FUNCTION(mysqli_stmt_data_seek)
 
 	MYSQLI_FETCH_RESOURCE_STMT(stmt, mysql_stmt, MYSQLI_STATUS_VALID);
 
-	mysql_stmt_data_seek(stmt->stmt, offset);
+	if (mysql_stmt_data_seek(stmt->stmt, offset)) {
+		zend_throw_error(NULL, "The result set buffer is empty");
```

Comment:
Not sure about the wording; is there even a buffer in this case. Maybe "Result set is not buffered" (or something like that) gives a better clue about what's wrong.

Reply:
Yes, currently nothing happens. This is because until now we supported libmysql, which as stated above has this function implemented as void. Now, we are no longer limited by libmysql and we can start throwing an error. 
I could also change this to a warning, but I don't see this happening very often and I don't it would be too much to have an error here. 

In regards to the text, I wanted to find something rather vague. The error could happen because someone forgot to call `store_result()`, the PS has only been prepared but not yet executed, or the PS didn't produce any result set. Your suggestion fits the first reason, but not so much the rest. I welcome some brainstorming here.